node/node.go

```diff
@@ -798,39 +811,48 @@ func (node *Node) Start() error {
 
 		// goroutine to handle node messages
 		go func() {
-			for m := range msgChanNode {
-				// should not take more than 10 seconds to process one message
-				ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
-				msg := m
-				go func() {
-					defer cancel()
-					if semNode.TryAcquire(1) {
-						defer semNode.Release(1)
-
-						if err := msg.handleE(ctx, msg.handleEArg, msg.actionType); err != nil {
-							errChan <- withError{err, nil}
+			for {
+				select {
+				case m := <-msgChanNode:
+					ctx, cancel := context.WithTimeout(node.psCtx, 10*time.Second)
+					msg := m
+					go func() {
+						defer cancel()
+						// AUDIT: DANGEROUS!! Some message might be dropped here!!!
+						// TODO: prevent using TryAcquire
+						if semNode.TryAcquire(1) {
```

Comment:
TryAcquire here is to prevent spamming attack.

Reply:
I See. Will remove the comment. BTW, do you think this may be the cause for low-performance machine cannot process all pub-sub messages?

Reply:
not just low performance, spamming p2p messages will cause OOM